website/docs/feature-flag-tutorials/rust/implementing-feature-flags-rust.md

```diff
@@ -0,0 +1,286 @@
+    // Convert to AVIF
+    let rgba_image = img.to_rgba8();
+    let pixels_rgba = rgba_image.as_raw().as_rgba();
+    let avif_data = AvifEncoder::new().with_quality(75.0).encode_rgba(Img::new(
+        pixels_rgba,
+        rgba_image.width() as usize,
+        rgba_image.height() as usize,
+    ))?;
+    std::fs::write("output.avif", avif_data.avif_file)?;
+
+    Ok(())
+}
+
+```
+
+Not as easy as the `webp` crate, I had to use the `rgb` crate to turn our image into a format that `ravif` would accept, but it works. Be mindful of the `usize` when passing the width and height.
+
+Run the code again:
+
+```sh
+cargo run
+```
+
+You should see another image named `output.avif` in your folder, alongside the `.webp` image. Make sure that all images are the same before continuing.
+
+## 4. Add Unleash to your Rust app
+
+Now, let’s connect our project to Unleash so that you can toggle that feature flag at runtime. If you wanted to, you could also do a gradual rollout, run A/B tests, etc.
+
+You’ll need 3 things:
+
+-   The Unleash SDK installed.
+-   The URL of your Unleash instance’s API. It’s `[http://localhost:4242/api/](http://localhost:4242/api/)` for your local version. You’ll want to replace it with your remote instance.
+-   The API token we created on our Unleash instance, feel free to create another one if you can’t find it.
+
+Let's first install all the SDK and all its dependencies.
+
+```sh
+cargo add enum-map@2.0.3
+cargo add reqwest --features json
+cargo add serde --features derive
+cargo add tokio --features full
+cargo add unleash-api-client --features reqwest
+```
+
+There are a few dependencies, and here's why: We need an HTTP client to make the request, and `serde` to deserialize the Unleash response. Our SDK is constantly polling Unleash to retrieve the value of feature flags, and caches that in memory. We don't want to block your app while this happens, so we need an async runtime. We're picking `tokio`.
+
+```rust
+use enum_map::Enum;
+use image::io::Reader as ImageReader;
+use ravif::{Encoder as AvifEncoder, Img};
+use rgb::FromSlice;
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+use tokio::time::sleep;
+use unleash_api_client::client;
+use webp::Encoder as WebPEncoder;
+
+#[derive(Debug, Deserialize, Serialize, Enum, Clone)]
+enum Flags {
+    avif,
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
+    let client = client::ClientBuilder::default().into_client::<Flags, reqwest::Client>(
+        "http://localhost:4242/api",
+        "unleash-rust-client-example",
+        "unleash-rust-client-example",
+        Some("default:development.unleash-insecure-api-token".to_string()),
+    )?;
+    client.register().await?;
+
+    let (_, _) = tokio::join!(client.poll_for_updates(), async {
+        // Ensure we have features for this demo.
+        sleep(Duration::from_millis(500)).await;
+
+        let avif = client.is_enabled(Flags::avif, None, false);
```

Comment:
No context here. I think that's okay for a demo but I would be a much happier ginger with a comment telling users that they should use one for production